presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1327,7 +1330,29 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 Optional.empty(),
                 discretePredicates,
                 ImmutableList.of(),
-                EMPTY_STATISTICS);
+                getTableStatistics(hivePartitions));
+    }
+
+    private TableStatistics getTableStatistics(List<HivePartition> hivePartitions)
+    {
+        List<Long> partitionRowNums = hivePartitions.stream()
+                .map(HivePartition::getStatistics)
+                .map(PartitionStatistics::getNumRows)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(toList());
+
+        long numRowsSum = partitionRowNums.stream().mapToLong(a -> a).sum();
+        long partitionsWithStatsCount = partitionRowNums.size();
+        long allPartitionsCount = hivePartitions.size();
+
+        TableStatistics tableStatistics = EMPTY_STATISTICS;
+        if (partitionsWithStatsCount > 0) {
+            tableStatistics = TableStatistics.builder()
+                    .setRowsCount(new StatisticsValue(1.0 * numRowsSum / partitionsWithStatsCount * allPartitionsCount, 0.0))
```

Comment:
I think we'd be better off not crossing the 120 chars per line limit (even if it's higher in presto codestyle) for better readability @ github